src/com/xilinx/rapidwright/util/VivadoTools.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Zak Nafziger, Advanced Micro Devices, Inc.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.xilinx.rapidwright.design.Design;
+
+/**
+ * Utility methods to run access vivado and parse logs
+ * 
+ * @author zakn
+ *
+ */
+public class VivadoTools {
```

Comment:
Is the idea to have multiple classes going forward dedicated to each task?  If so, I would suggest that we create a dedicated package for Vivado instead of putting them all into the same class.  We have tried to avoid inner classes because they tend to complicate compilation and a few other areas.

Alternatively, the class could be converted into a static method and just return a `Map<String,Integer>` that contains the results.  This would be simpler and would probably be the preferred method in my opinion.

Reply:
@eddieh-xlnx has been directing some of the architecture decisions here, but I think the idea is to add classes for future vivado tasks, so maybe a separate package is warranted.

Returning a `Map<String,Integer>` is perhaps little more awkward, since the user then needs to know exactly what vivado output log key phrase they want (i.e. `"# of unrouted nets"`). I was mostly following the proposed system from here: https://github.com/Xilinx/RapidWright/pull/684#discussion_r1205979575

Reply:
> @eddieh-xlnx has been directing some of the architecture decisions here, but I think the idea is to add classes for future vivado tasks, so maybe a separate package is warranted.

Ok, sure.  Let's keep it the way it is for now and we can revisit it in the future if the file seems becomes unwieldly.  

> Returning a `Map<String,Integer>` is perhaps little more awkward, since the user then needs to know exactly what vivado output log key phrase they want (i.e. `"# of unrouted nets"`). I was mostly following the proposed system from here: [#684 (comment)](https://github.com/Xilinx/RapidWright/pull/684#discussion_r1205979575)

Sure, I can see how that is a bit cumbersome and error prone.  For simplicity, let's keep it as is.